src/List.mo

```diff
@@ -252,50 +547,115 @@ module {
 
   /// Merge two ordered lists into a single ordered list.
   /// This function requires both list to be ordered as specified
-  /// by the given relation `lte`.
-  public func merge<T>(l1 : List<T>, l2 : List<T>, lte : (T, T) -> Bool) : List<T> {
+  /// by the given relation `lessThanOrEqual`.
+  ///
+  /// Example:
+  /// ```motoko include=initialize
+  ///
+  /// List.merge<Nat>(
+  ///   ?(1, ?(2, ?(4, null))),
+  ///   ?(2, ?(4, ?(6, null))),
+  ///   func (n1, n2) { n1 <= n2 }
+  /// ); // => ?(1, ?(2, ?(2, ?(4, ?(4, ?(6, null))))))),
+  /// ```
+  ///
+  /// Runtime: O(size(l1) + size(l2))
+  ///
+  /// Space: O(size(l1) + size(l2))
+  ///
+  /// *Runtime and space assumes that `lessThanOrEqual` runs in O(1) time and space.
+  // TODO: replace by merge taking a compare : (T, T) -> Order.Order function?
+  public func merge<T>(l1 : List<T>, l2 : List<T>, lessThanOrEqual : (T, T) -> Bool) : List<T> {
     switch (l1, l2) {
       case (null, _) { l2 };
       case (_, null) { l1 };
       case (?(h1, t1), ?(h2, t2)) {
-        if (lte(h1, h2)) {
-          ?(h1, merge<T>(t1, l2, lte))
+        if (lessThanOrEqual(h1, h2)) {
+          ?(h1, merge<T>(t1, l2, lessThanOrEqual))
         } else {
-          ?(h2, merge<T>(l1, t2, lte))
+          ?(h2, merge<T>(l1, t2, lessThanOrEqual))
         }
       }
     }
   };
 
-  /// Compare two lists using lexicographic ordering specified by the given relation `lte`.
-  public func compare<T>(l1 : List<T>, l2 : List<T>, compElm : (T, T) -> Order.Order) : Order.Order {
+  /// Compare two lists using lexicographic ordering specified by the given relation `compareElem`.
+  ///
+  /// Example:
+  /// ```motoko include=initialize
+  /// import Nat "mo:base/Nat";
+  ///
+  /// List.compare<Nat>(
+  ///   ?(1, ?(2, null)),
+  ///   ?(3, ?(4, null)),
+  ///   Nat.compare
+  /// ) // => #less
+  /// ```
+  ///
+  /// Runtime: O(size(l1))
+  ///
+  /// Space: O(1)
+  ///
+  /// *Runtime and space assumes that `compareElem` runs in O(1) time and space.
+  public func compare<T>(l1 : List<T>, l2 : List<T>, compareElem : (T, T) -> Order.Order) : Order.Order {
```

Comment:
```suggestion
  public func compare<T>(l1 : List<T>, l2 : List<T>, compare : (T, T) -> Order.Order) : Order.Order {
```
Is this possible? Or does the shadowing not allow it

Reply:
For consistency with other modules

Reply:
It would be possible by adding a private auxilliary function and calling that.

Reply:
Done